src/hooks/useQuizState/useQuizApiState.ts

```diff
@@ -103,8 +120,13 @@ const useQuizApiState: UseQuizApiState = (
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [quizApiState.resultsConfig]);
 
+  const prevQuizId = usePrevious(quizId);
+
   useEffect(() => {
     (async () => {
+      // If quizId is the same as the previous quizId, wait for next render after reset
+      if (!!prevQuizId && quizId !== prevQuizId) return;
```

Comment:
Would you be able to elaborate a bit more on what this is for and what it's doing? Thanks!

Reply:
+1 on Jimmy's ask. Why are we keeping track of a `prevQuizId` state?

Reply:
There was an issue with merging my remote branches. Removed here [207f8fe](https://github.com/Constructor-io/constructorio-ui-quizzes/pull/216/commits/207f8fe483a6b125eb22caea2cf67a047b597137)